### PR TITLE
planner: Add min max subquery for partition pruning

### PR DESCRIPTION
**Pull request by elsa0520**
Add min subquery for partition pruning

### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/rule_predicate_push_down.go [hunk 1]
```diff
@@ -17,7 +17,14 @@ package core
 import (
 	"bytes"
 	"context"
+	"errors"
 	"fmt"
+	"github.com/pingcap/tidb/expression/aggregation"
```
Comment (AilinKid) [inline planner/core/rule_predicate_push_down.go:22]: group import

**Review by fixdb — DISMISSED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**